privly.safariextension/scripts/content_scripts/privly.js

```diff
@@ -395,6 +400,14 @@ var privly = {
 
     //put the iframe into the page
     object.parentNode.insertBefore(iFrame, object);
+
+    // Give the iframe 2s to load. If CSP blocks iframe injection then,
+    // stop the content script.
+    setTimeout(function() {
+      if (privly.blockedByCSP === true) {
+        privly.stop();
+      }
+    }, 2000);
```

Comment:
I am saying privly.js should not make any changes to the host page aside from testing whether we can inject the app. If the test is successful, then we start changing the links. The `CSP_iframe.html` should not be visible to the user during the test.